### PR TITLE
Docs/user guide for web-app-react

### PR DESCRIPTION
**Pull request by andreasrs**
First version of a comprehensive user-guide for our lovely users 👥 

Todo:
- [x] Collect feedback on content + adjust
- [x] Make this GUIDE.md somehow be included in projects when running `roc docs`

### CODE REVIEW

File: GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,1021 @@
...
+
+`web-app-react` does not add any polyfills for you. If you wish to add custom polyfills this can be done by intalling the required polyfills to your project. For example:
+
+`$ npm i --save core-js`
+
+Add a file that can group your polyfilling ambitions, `src/polyfills.js` to the project:
+
+```javascript
+// only include polyfills on the client
+if (__WEB__) {
```
Comment (vasklund) [inline GUIDE.md:986]: Is this for illustrative purposes or is it something the stack exposes for us?

**Comment by vasklund** [comment]
As a general feedback, maybe the guide should be split into several pages? It makes it easier to consume in bit-sized pieces :)

**Comment by runeh** [comment]
Personal opinion: Having a single document is way easier to deal with, since it means I can search in the document.

File: GUIDE.md (hunk 1 above)
Comment (andreasrs) [reply GUIDE.md:986]: Actually this is available for us in the project and can be used for things like this.

I'll add a section about that, as it is pretty useful 👍

**Comment by andreasrs** [comment]
I think we will stick with a single document for now and then re-evaluate if it turns out to be an issue

File: GUIDE.md (hunk 1 above)
Comment (andreasrs) [reply GUIDE.md:986]: Added here https://github.com/rocjs/roc-package-web-app-react/pull/78/commits/9ff220197d8af4e8f2bd96db513fac22768861c4

File: GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,1081 @@
+# User-guide
+
+`roc-package-web-app-react` provides a complete _framework_ for creating and running ambitious Node.js and React web applications. It is developed by [Schibsted](http://www.schibsted.com/en/About-Schibsted/Schibsted-Products-and-Technology/) and used in production across many products.
+
+It is a perfect match for writing fully-featured server-rendered web applications that use `React` components and `Redux` to handle your flow of data. It does not compromise on features or flexibility. Plugins are available for all important features like styling pipelines and automated testing.
```
Comment (dlmr) [inline GUIDE.md]: What about adding links to React and Redux here, and not making them code-blocks?

Comment (dlmr) [inline GUIDE.md]: We mention plugins here without further explanation, should we maybe add a link to make it easier to understand what plugins we are talking about here.

File: GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,1081 @@
...
+
+You need `Node.js` 4.x+ and `npm` 3.x+ to use Roc. You can install Roc globally on your system using `npm`.
+
+`$ npm i -g roc@next`
+
+## Creating your app:
+
+Bootstrapping a new project in the location of your choice is fully automated. Replace `my-app` in the command below with your preferred name.
+
+`$ roc new my-app web-app-react wip`
```
Comment (dlmr) [inline GUIDE.md]: Add an explanation what `wip` is here maybe?

File: GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,1081 @@
...
+General options:
+ -b, --better-feedback                   Enables source-map-support and loud-rejection.
+ -c, --config                            Path to configuration file.
+ -d, --directory                         Path to working directory.
+ -h, --help                              Output usage information.
+ -V, --verbose                           Enable verbose mode.
+ -v, --version                           Output version number.
+```
+
+We used `new` from the `project creation` group when bootstrapping our project in the previous section. What deserves our attention now are the additional commands grouped under `project development` as well as `start`:
```
Comment (dlmr) [inline GUIDE.md]: > the `project creation` group

Change that into `Project creation`?

Comment (dlmr) [inline GUIDE.md]: > under `project development` as well as `start`:

 under `Project development` as well as the `start` command:

File: GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,1081 @@
...
+Project development:
+ development build [targets]             Build the current project.
+ development clean                       Cleans the current project.
+ development dev [targets]               Starts the current project in development mode.
+ development test [targets]              Runs tests on the current project.
+```
+
+These commands are important for your daily development workflow and they can be abbreviated as `roc build`, `roc clean`, `roc dev` and `roc test`. For information about how to use the commands and their available settings add the help option `-h` (`--help`) when running them.
+
+Your bootstrapped project has also created command aliases in your `package.json` for these commands, so `npm run build`, `npm start` etc. also work as expected for those already familiar with npmscripts.
```
Comment (dlmr) [inline GUIDE.md]: > npmscripts

npm scripts

File: GUIDE.md [hunk 6]
```diff
@@ -0,0 +1,1081 @@
...
+
+To start the development server and launch the app in your browser, run  
+`$ roc dev` which will open a new tab like in the screenshot:
+
+![my-app in the browser](rocdev.png)
+
+Here you see your React app running in development mode. You can open the files in your favourite code editor and they will be live-reloaded in the browser as you make modifications to them. Development mode in your Roc project is powerful and you get a great set of tools that are ready to be used. No extra setup required.ß
```
Comment (dlmr) [inline GUIDE.md]: > No extra setup required.ß

Remove the `ß`. 🙂

File: GUIDE.md [hunk 7]
```diff
@@ -0,0 +1,1081 @@
...
+- Host of backend API servers to use
+- Feature flags for the application
+- Language settings
+
+Hardcoding configuration values like this that can vary in different environments is not a great idea.
+
+To solve this in a good manner, `web-app-react` includes and exposes [node-config](https://github.com/lorenwest/node-config) to your project. It takes care of all your configuration needs.
+
+### Defining runtime configuration with node-config
+Create a new directory `config/` in your project root. Here you may define `js` or `json` files named like the corresponding environment (`NODE_ENV`) of choice. For example:
```
Comment (dlmr) [inline GUIDE.md]: You can define many other formats as well, like YAML.

File: GUIDE.md [hunk 8]
```diff
@@ -0,0 +1,1081 @@
...
+);
+```
+
+You can consult the [official docs for React](https://facebook.github.io/react/docs/components-and-props.html) on components and props.
+
+## Styling React components
+
+Roc does not enforce a particular way of styling components on you. We do however give a fully working styling pipeline as a part of the default template used in this guide; provided by `roc-plugin-style-sass`. This plugin can be found in your `package.json` and removed if you wish to replace it with something else later.
+
+`roc-plugin-style-sass` is a very powerful and flexible way of handling styles in your app. It uses [sass](http://sass-lang.com/) for pre-processing and also includes [css-module](https://github.com/css-modules/css-modules) support out of the box.
```
Comment (dlmr) [inline GUIDE.md:326]: Link to the alternative we have, less?

File: GUIDE.md [hunk 9]
```diff
@@ -0,0 +1,1081 @@
...
+```
+
+This defaults to ``"src/template-values.js"``. In this file you can define a function that returns an object with values that will be exposed in the template as `custom`.
+
+Example `src/template-values.js`:
+```javascript
+export default function getTemplateValues({ koaState, settings, reduxState }) {
+  // you can use state from koa, settings and redux state to make your decision
+  return {
+      bodyClass: 'main',
```
Comment (dlmr) [inline GUIDE.md:433]: This is a somewhat undocumented feature, should we perhaps use something else for the example?

File: GUIDE.md [hunk 10]
```diff
@@ -0,0 +1,1081 @@
...
+- Trigger new data fetches upon route transitions (for example, the user navigates from `/` to `/data`)
+- Do data fetching on the server so that we do not always have to wait on the client for shared/common data
+- Do data fetching on the client only, for personalized content and similar things that are unique to the user
+- Choose to do data fetching before or after a route transition has taken place on the client
+
+To achieve all of this, `web-app-react` provides [react-router-redial](https://github.com/dlmr/react-router-redial).  
+It exposes an API that allows us to enhance our components rendered by `react-router` with dedicated data-fetching capabilities.
+
+Using it is simple:
+- In your component; import `provideHooks` from `react-router-redial`
```
Comment (dlmr) [inline GUIDE.md]: from `redial`.

File: GUIDE.md [hunk 11]
```diff
@@ -0,0 +1,1081 @@
...
+      <div />
+    );
+  }
+}
+```
+
+If you answered `Y` to include a Redux demo in your app you can view some example actions in the bottom of `src/redux/repo.js`
+
+### Redux reducers
+Roc bootstraps Redux for us and expects to find your reducers exported in the file configured by
```
Comment (dlmr) [inline GUIDE.md]: Should this be higher? Without a reducers file we don't get Redux in the application.

File: GUIDE.md [hunk 12]
```diff
@@ -0,0 +1,1081 @@
...
+- [YellowBox](https://github.com/iamdustan/yellowbox-react) is installed and active for notifying about potential issues
+- [Redux devtools](https://github.com/gaearon/redux-devtools) are available on `ctrl+h` by default
+- [Browsersync](https://www.browsersync.io/) admin tool is running on port 3031 by default
+- React is in development mode for better feedback
+- It enables hot-reloading of your code
+- It is unsuitable for production, in every way imaginable
+
+Never run `dev` in a production environment. It is tailored for a good development experience but is not suited for production traffic.
+
+Use `roc start` to serve your production-grade bundles.
```
Comment (dlmr) [inline GUIDE.md:1118]: Give an example on how it's possible to create a custom entry point, for when using together with something like a logger on the server.

**Review by dlmr — COMMENTED** [review]
Great stuff!

The only thing I would change if possible other than my comments here is to add code examples directly inline over referring to the result from the template generation, making everything stand a little more on its own.

File: GUIDE.md (hunk 3 above)
Comment (andreasrs) [reply GUIDE.md]: Sure. Was hoping we could merge this https://github.com/rocjs/roc-template-web-app-react/pull/15 and use the default instead.

File: GUIDE.md (hunk 7 above)
Comment (andreasrs) [reply GUIDE.md]: Sure. Can link to the appropriate docs.

File: GUIDE.md (hunk 8 above)
Comment (andreasrs) [reply GUIDE.md:326]: Do we want to advertise that? I am unsure if it is in a suitable state as I have not used it much myself. If it's all good I'll throw in a link to it.

File: GUIDE.md (hunk 9 above)
Comment (andreasrs) [reply GUIDE.md:433]: Sure. Suggestions on a suitable alternative?

Comment (andreasrs) [reply GUIDE.md:433]: Adding a note that it will likely be removed in the future.

**Comment by andreasrs** [comment]
Sweet feedback @dlmr 🎉 

All suggestions should be implemented now.

**Review by dlmr — APPROVED** [review]
👍  Remember to squash before merge.

**Comment by andreasrs** [comment]
👍 Leaving the branch for a little while because of the direct-links that have been shared

**Pull request merged**